File: GTP2.5/gtp4.cpp
#include<iostream>
#include<string.h>

using namespace std;

int count_vists(int visits[]) {
        int count = 0;

        for(int i = 0; i < 8; i++) {
                count += (visits[i] > 4);
        }

        return count;
}

void sort(string names[], int ages[]) {
        int b = 0, tmp_n;
        string tmp_s;

        while(b == 0) {
                b = 1;

                for(int i = 0; i < 7; i++) {
                        if(ages[i] > ages[i+1]) {
                                b = 0;

                                tmp_n = ages[i];
                                ages[i] = ages[i+1];
                                ages[i+1] = tmp_n;

                                tmp_s = names[i];
                                names[i] = names[i+1];
                                names[i+1] = tmp_s;
                        }
                }
        }
}

int main() {
        string names[8] = {"Martin", "Germán", "León", "Moisés", "Andréas", "John", "Pepe", "Alberto"};
        int ages[8] = {21, 22, 51, 18, 34, 12, 77, 56};
        int visits[8] = {1,2,3,5,6,5,1,6};

        cout << "La cantidad de pacientes con más de 4 visitas es: " << count_vists(visits) << "\n";

        sort(names, ages);

        cout << "Los pacientes mas longevos son: " << names[5] << ", " << names[6] << ", " << names[7];
        cout << "\nEl más joven: " << names[0] << "\n";
        
        return 0;
}

File: GTP2.5/gtp4.py
def ordenar(ed:list[int], nom:list[str]):
        b = 0

        while(b == 0):
                b = 1

                for i in range(len(ed) - 1):
                        if ed[i] < ed[i+1]:
                                aux = ed[i]
                                ed[i] = ed[i+1]
                                ed[i+1] = aux
                
                                aux = nom[i]
                                nom[i] = nom[i+1]
                                nom[i+1] = aux

                                b = 0

def main():
        S = 3

        nombre = []
        edad = []
        vist = []
        contador = 0

        for i in range(S):
                nombre.append(input(f"Ingrese el nombre del paciente {i+1}: "))
                edad.append(int(input("Ingrese su edad: ")))
                vist.append(int(input("Visitas previas: ")))
                contador += vist[i] > 4
                print()

        ordenar(edad, nombre)
        
        print(f"La cantidad de pacientes con más de 4 visitas previas es: {contador}")
        print(f"Los pacientes más longevos son: {nombre[0]}, {nombre[1]}, {nombre[2]}")
        print(f"El paciente más jóven atendido en el día es: {nombre[S-1]}")

if __name__ == '__main__':
        main()
